docs/dev/plugins/cloudformation/architecture.md

```diff
@@ -0,0 +1,47 @@
+![Class diagram of the cloudformation package](img/architecture-lucidchart.png)
+
+## Cloudformation package
+
+The `cloudformation` package is the basic package of the CloudFormation plugin. It contains the `CloudFormationPlugin`, `CloudFormationLifecycle`, `CloudFormationModule` and `CloudFormationFileCreator` classes.
+
+### `CloudFormationPlugin`
+
+`CloudFormationPlugin` is an extension of the `ToscanaPlugin` class representing the CloudFormation platform. The platform properties of the CloudFormation plugin are the `AWS Region` which determines where the transformed model will run in, the `AWS Access Key ID` and the `AWS Secret Key` which are the credentials used to communicate with AWS during the transform phase of the transformation lifecycle and the `AWS Keypair` which, if set to true, makes sure that the EC2 instances can be accessed via SSH with a valid [AWS EC2 Keypair](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html).
+
+> **Note**: If the `AWS KeyPair` property is set to **true**, the name of your AWS EC2 Keypair must be manually replaced in the `create-stack.sh` script of the [target artifact](deployment/target-artifact.md#scripts) before deployment.
+
+### `CloudFormationLifecycle`
+
+`CloudFormationLifecycle` is an extension of the `AbstractLifecycle` class which contains the lifecycle phases that are executed if the user initiates the transformation procedure.
+When the lifecycle is constructed, the `CloudFormationModule` is initiated with the `fileAccess` and platform properties from the `TransformationContext` which is passed to the lifecycle as a parameter. The `model` field is also set with `EffectiveModel` from the `TransformationContext` which is later used during the lifecycle phases as a basis for the transformation.
+For further information about the behaviour of the CloudFormation plugin during each of these phases, please refer to the [transformation workflow](transformation/transformation-workflow.md).
+
+### `CloudFormationModule`
+
+`CloudFormationModule` is an extension of the [`Module`](https://github.com/StuPro-TOSCAna/cloudformation-builder/blob/master/src/main/java/com/scaleset/cfbuilder/core/Module.java) class of the cloudformation-builder. Originally, it was used in order to represent the Java model of a CloudFormation template, but in order to fulfil the needs of our plugin, we extended it with additional fields and methods needed to facilitate the deployment of our template. During the transformation, it gets filled with the information in order to create the CloudFormation template and any additional files needed for deployment.
+
+## Visitor package
+
+The `visitor` is used to iterate over the EffectiveModel during the TransformationLifecycle. It contains the `CloudFormationVisitor`, `CheckModelNodeVisitor`, `CheckModelRelationshipVisitor`, `PrepareModelNodeVisitor`, `PrepareModelRelationshipVisitor` and `TransformModelNodeVisitor` classes.
```

Comment:
I'd not list the content of the package. This is a maintenance burden, as it will become outdated as soon as someone introduces a new listener.